### ISSUE TITLE
Update and delete fail with a Lightswitch odata service

### ISSUE THREAD

**Issue by Anton8** [object/Simple.OData.Client.xamstore#1]
I am trying to do a hello world CRUD with my authenticated lightswitch odata service from Xamarin on Android.

The client worked perfectly for inserting and getting data, but failed for the delete and update routines. (just hangs)

I noted that someone else had an issue with authenticated access (https://github.com/object/Simple.OData.Client/issues/23).  He used some sort of delegate with the BeforeRequest  method to get it to work. Is there somewhere that explains this a little more as I would like to try this approach to see if it works. Or can you suggest any other options.  Any suggestion much appreciated, thanks.

Currently my code looks like:

```
        var settings = new ODataClientSettings();
        settings.UrlBase = "https://tmalive.azurewebsites.net/xxxLiveData.svc/";
        settings.Credentials = new System.Net.NetworkCredential("Admin", "xdata!");
        var client = new ODataClient(settings);
        String returnValue = "";


        // DELETE NOT WORKING
        RunOnUiThread(() => phoneNumberText.Text = "Trying to delete existing record.");
        await client
            .For("JobTypes")
            .Key(7)
            .DeleteEntryAsync();

        returnValue = "Deleted OK";
        RunOnUiThread(() => phoneNumberText.Text = returnValue);
```

**Comment by object** [comment]
Hello Anton,

The library has a set of tests, including tests of insert/update/delete agains the Azure Web site using basic authentication. You can check them here: https://github.com/object/Simple.OData.Client/blob/master/Simple.OData.Client.Tests.Net40/WebApiTests.cs. Your code looks fine to me. Did you try your code in a desktop app? Often such problems are caused by deadlocks. What is your platform?

Vagif

**Comment by Anton8** [comment]
Hi Vagif,
Thanks for your swift response.
My platform is VS2012, windows 7 coding for Android 4.1 (API 16).
It shouldn't be a deadlock as I'm only doing one activity at a time, i.e. either add or update or read or delete.
I'll try the code from a desktop app.  I'll also try the test code - is that odata service operational?

cheers
Anton

**Comment by Anton8** [comment]
HI Vagif,

I have run the tests from a desktop app - and they work fine.

I have re-run my code on a desktop app and the error it gave me was:

A first chance exception of type 'System.ArgumentNullException' occurred in mscorlib.dll 

It occurred on the line (.updateEntryAsync();)

await client
                .For("JobTypes")
                .Key(7)
                .Set(new { JobType1 = newJobTypeName, IsActive = false })
                .UpdateEntryAsync();

I believe there is a problem handling the Lightswitch odata feed.  A few years back I was trying to get Android to read a Lightswitch Odata feed and I starting using Odata4j open source library.  Exactly the same problem arose: Read/Create worked fine, but Update/Delete failed.

I am guessing there is something slightly different in the Lightswitch oData service format which causes the problem.  If it is helpful to have the metadata, I can provide it.  If you think it could be fixed I can organise access for you to the service.

If there's anything I can do, please tell me.

Kind regards,

**Comment by object** [comment]
Hi Anton,

The most efficient for me is if I can get temporary access to your service. Then I can debug directly from my machine. You can use my email address vagif.abilov@gmail.com if you need to send me credentials etc.

Cheers

**Comment by Anton8** [comment]
Hi Vagif - thanks very much - Have sent the details through to you.
Kind regards,
Anton

**Comment by Anton8** [comment]
Hi Vagif,

Please let me know when the updated release is available and I will test it immediately.
Kind regards,

Anton

**Comment by Anton8** [comment]
Found it, thanks.  Testing now.  Will let you know ASAP

**Comment by Anton8** [comment]
Nope - wrong one that was Release rc1.  Will await release rc2.  Thanks

**Comment by object** [comment]
It will come in RC2, but as I wrote you in my direct mail, I have to resolve issues caused by this fix. Hopefully by tomorrow.

**Comment by object** [comment]
Resolved in the version 3.1.
